samcli/commands/sync/command.py

```diff
@@ -415,3 +419,20 @@ def execute_watch(
     """
     watch_manager = WatchManager(template, build_context, package_context, deploy_context, auto_dependency_layer)
     watch_manager.start()
+
+
+def check_enable_dependency_layer(template_file: str):
+    """
+    Check if auto dependency layer should be enabled
+    :param template_file: template file string
+    :return: True if ADL should be enabled, False otherwise
+    """
+    stacks, _ = SamLocalStackProvider.get_stacks(template_file)
+    for stack in stacks:
+        esbuild = EsbuildBundlerManager(stack)
+        if esbuild.esbuild_configured():
+            # Disable ADL if esbuild is configured. esbuild already makes the package size
+            # small enough to ensure that ADL isn't needed to improve performance
+            click.secho("esbuild is configured, disabling auto dependency layer.", fg="yellow")
```

Comment:
discussed through huddle: potentially display some explanation to users on why ADL is disabled for esbuild?